skbio/alignment/_tabular_msa.py

```diff
@@ -837,6 +838,148 @@ def consensus(self):
         return dtype(''.join(consensus),
                      positional_metadata=positional_metadata)
 
+    def _build_inverse_shannon_uncertainty_f(self, include_gaps):
+        base = len(self.dtype.nondegenerate_chars)
+        if include_gaps:
+            # Increment the base by one to reflect the possible inclusion of
+            # the default gap character.
+            base += 1
+
+        def f(p):
+            freqs = list(p.kmer_frequencies(k=1).values())
+            return 1. - scipy.stats.entropy(freqs, base=base)
+        return f
+
+    @experimental(as_of='0.4.0-dev')
+    def conservation(self, metric='inverse_shannon_uncertainty',
+                     nan_on_degenerate=False, gap_mode='nan'):
+        """Apply metric to compute conservation for all alignment positions
+
+        Parameters
+        ----------
+        metric : str, optional, {'inverse_shannon_uncertainty'}
+            Metric that should be applied for computing conservation. Resulting
+            values should be larger when a position is more conserved.
+        nan_on_degenerate : bool, optional
```

Comment:
Update this to be ``degenerate_mode`` currently supporting only ``nan`` and ``error``.